### PR TITLE
OPEX-929: Add an entry for Datadog under the Log Streams documentation

### PR DESCRIPTION
**Pull request by Deborah-Digges**
# Description

Adds a page for the Datadog Log Streams option.

# Reference
https://auth0team.atlassian.net/browse/OPEX-929

# Feature Demo

https://drive.google.com/file/d/1j2-jDWMXYHCCk9sMK3EiOV0v9j1Hb0EV/view?usp=sharing

### CODE REVIEW

File: articles/logs/streams/datadog.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
```
Comment (Deborah-Digges) [inline articles/logs/streams/datadog.md:1]: Raising this PR for initial feedback, but we will probably merge it in a week or so when the Datadog team is ready on their side with the integration:

https://docs-content-staging-pr-8792.herokuapp.com/docs/logs/streams/datadog

**Review by jeffreylees — DISMISSED** [review]

**Comment by jeffreylees** [comment]
Reviewed and ready to merge when the integration is.

File: articles/logs/streams/datadog.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+topics:
+ - logs
+ - streams
+ - event-streams
+contentType: how-to
+---
+
+# Datadog Event Log Streams
+
+Datadog Event Log Streams let you export your log events to Datadog for further analysis and monitoring. When Auth0 creates a log entry for your tenant, a copy is automatically sent to Datadog with the `source` and `service` both set to `auth0`.
```
Comment (camicano) [inline articles/logs/streams/datadog.md]: For consistency with the eventbridge docs I think this paragraph should give an overview of the value of the integration. 

Datadog is the essential monitoring platform for cloud applications. That brings together together data from servers, containers, databases, and third-party services to make your stack entirely observable.

You can create monitoring, alerting and analysis dashboards for Auth0 tenants.

File: articles/logs/streams/datadog.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## Create a Datadog Event Stream
+
+### Copy API Key from Datadog
+
+1. Head to the Datadog dashboard.
```
Comment (camicano) [inline articles/logs/streams/datadog.md]:
```suggestion
1. Log in to the Datadog dashboard.
```

File: articles/logs/streams/datadog.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+2. Navigate to **Integrations** -> **APIs**.
+![Integrations Dashboard](/media/articles/logs/datadog/tutorial-1.png)
+3. Expand the API Keys section on this page, then copy the API Key that you would like to use.
+![API Keys Section](/media/articles/logs/datadog/tutorial-2.png)
+
+### Configure settings in Auth0
```
Comment (camicano) [inline articles/logs/streams/datadog.md]:
```suggestion
### Set up log stream in Auth0
```

File: articles/logs/streams/datadog.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+1. Log in to the [Auth0 Dashboard](${manage_url}).
+2. Navigate to **Logs > Streams**.
+3. Click **+ Create Stream**.
+4. Select **Datadog** and enter a unique name for your new Datadog Event Stream.
+5. On the next screen, provide the following settings for your Datadog Event Stream:
```
Comment (camicano) [inline articles/logs/streams/datadog.md:40]: I think step 5 should be more consistent to how we explain it for Eventbridge

**Review by camicano — CHANGES_REQUESTED** [review]
In general I think we should make this page and the eventbridge docs more consistent but other than that is looks good!

File: articles/logs/streams/datadog.md (hunk 2 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md]: Added paragraph about Datadog value

File: articles/logs/streams/datadog.md (hunk 3 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md]: Done

File: articles/logs/streams/datadog.md (hunk 4 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md]: Done

File: articles/logs/streams/datadog.md (hunk 5 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md:40]: @camicano , I need to provide a little more explanation about each field here which is why I need the table. Is that ok?

Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md:40]: Also, this is consistent with the HTTP docs:

https://docs-content-staging-pr-8792.herokuapp.com/docs/logs/streams/http-event

**Review by camicano — DISMISSED** [review]

File: articles/logs/streams/datadog.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+### View logs in Datadog
+
+1. Navigate to **Logs** > **Livetail**.
+2. See Auth0 logs by setting the `source` to `auth0`.
+![Datadog Logs Dashboard](/media/articles/logs/datadog/tutorial-4.png)
+
+## Delivery attempts and retries
+
+Auth0 events are delivered to your server via a streaming mechanism that sends each event as it is triggered. If your server is unable to receive the event, Auth0 will try to redeliver it up to three times. If still unsuccessful, Auth0 will log the failure to deliver.
```
Comment (sandrinodm) [inline articles/logs/streams/datadog.md]: How do retries work? Are they retried immediately? Or is it after 1 hour, then after a day for example?

How can I troubleshoot retries/delivery issues?

File: articles/logs/streams/datadog.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+| Setting | Description |
+|---------|-------------|
+| API Key | The Datadog API key you copied from the Datadog dashboard. |
+| Region  | If you are in the Datadog EU site (app.datadoghq.eu), the `Region` should be `EU`; otherwise, it should be `US`. |
+
+![Datadog Settings Form](/media/articles/logs/datadog/tutorial-3.png)
+
+6. Click **Save**.
+7. You're done! When Auth0 writes the next tenant log, you'll receive a copy of that log event in Datadog with the `source` and `service` set to `auth0`.
```
Comment (sandrinodm) [inline articles/logs/streams/datadog.md]: I would rename "tenant log" to something else. From our point of view these are tenants and their tenant logs (as opposed to application logs).

But as a customer these are just logs or log events.

What about `When Auth0 writes the next log event`

File: articles/logs/streams/datadog.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+topics:
+ - logs
+ - streams
+ - event-streams
+contentType: how-to
+---
+
+# Datadog Event Log Streams
+
+Datadog is the essential monitoring platform for cloud applications. It brings together data from servers, containers, databases, and third-party services to make your stack entirely observable.
```
Comment (sandrinodm) [inline articles/logs/streams/datadog.md:14]: What Datadog subscription do I need to leverage this? Is it Datadog Log Management? Or Datadog Infrastructure? We should clarify.

https://www.datadoghq.com/pricing/

File: articles/logs/streams/datadog.md (hunk 7 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md]: That's an excellent point. I've updated the docs accordingly. Thank you for your feedback!

File: articles/logs/streams/datadog.md (hunk 6 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md]: We didn't want to explain too much about our implementation of the retry mechanism to customers.

However, we have implemented a `Health` tab this quarter(which we are also releasing today) which will allow them to troubleshoot these issues.

I have added a small note about it here as well 👍

File: articles/logs/streams/datadog.md (hunk 8 above)
Comment (Deborah-Digges) [reply articles/logs/streams/datadog.md:14]: Excellent point 👍 I've added a note about the plan they need in order for this integration to work

**Review by rkhoriander — APPROVED** [review]

**Pull request merged**